src/os_posix/os_alloc.c

```diff
@@ -58,7 +58,8 @@ __wt_calloc(WT_SESSION_IMPL *session, size_t number, size_t size, void *retp)
 		WT_STAT_FAST_CONN_INCR(session, memory_allocation);
 
 	if ((p = calloc(number, size)) == NULL)
-		WT_RET_MSG(session, __wt_errno(), "memory allocation");
+		WT_RET_MSG(session, __wt_errno(), "memory allocation of %"
+		    PRIu64 " bytes failed", size * number);
```

Comment:
Also, I know what you've got is syntactically correct, but please get the "%" on the same line as the format specifier.